File: Problems/Algorithms/242. Valid Anagram/ValidAnagram.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        return counter(s) == counter(t)
    }
    
    private fun counter(s: String): MutableMap<Char, Int> {
        val map: MutableMap<Char, Int> = mutableMapOf<Char, Int>()
        s.forEach { map.put(it, map.getOrDefault(it, 0) + 1) }
        return map
    }
}

File: Problems/Algorithms/242. Valid Anagram/valid-anagram.cpp
#include <string>

using namespace std;

class Solution {
private:
    int map[26] = {0};

public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }

        for (int i = 0; i < s.size(); i++) {
            map[s[i]-'a']++;
        }
        for (int j = 0; j < t.size(); j++) {
            map[t[j]-'a']--;
        }

        for (int k = 0; k < 26; k++) {
            if (map[k] != 0) {
                return false;
            }
        }
        return true;
    }
};
